### ISSUE TITLE
[operator] NoClassDefFoundError: com/fasterxml/jackson/databind/Module at runtime

### ISSUE THREAD

**Issue by agemooij** [lightbend/cloudflow#43]
**Describe the bug**
After building the operator docker image, pushing it to a repo, and adjusting the installation script to use it, Gitter user @mrooding ran into a runtime error: `java.lang.NoClassDefFoundError: com/fasterxml/jackson/databind/Module`

**To Reproduce**
- build the operator docker image using `sbt docker` in the operator sub-project
- tag and push the resulting image to a registry
- adjust the installation script to use the above image
- install

Check the logs

**Expected behavior**
No errors

**Screenshots**

**Additional context**

**Comment by skonto** [comment]
I was able to reproduce this by building a custom imag and trying to deploy Cloudflow:
```
2019-11-29 17:45:18,117 INFO  [ActorSystemImpl] - Connected to Kubernetes cluster: https://10.3.240.1:443
Uncaught error from thread [default-akka.actor.default-dispatcher-2]: com/fasterxml/jackson/databind/Module, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[default]
java.lang.NoClassDefFoundError: com/fasterxml/jackson/databind/Module
	at play.api.libs.json.StaticBinding$.parseJsValue(StaticBinding.scala:16)
	at play.api.libs.json.Json$.parse(Json.scala:171)
	at skuber.json.PlayJsonSupportForAkkaHttp.$anonfun$unmarshaller$2(PlayJsonSupportForAkkaHttp.scala:83)
	at akka.http.scaladsl.util.FastFuture$.$anonfun$map$1(FastFuture.scala:23)
	at akka.http.scaladsl.util.FastFuture$.strictTransform$1(FastFuture.scala:41)
	at akka.http.scaladsl.util.FastFuture$.transformWith$extension1(FastFuture.scala:45)
	at akka.http.scaladsl.util.FastFuture$.map$extension(FastFuture.scala:23)
	at akka.http.scaladsl.unmarshalling.Unmarshaller.$anonfun$map$3(Unmarshaller.scala:26)
	at akka.http.scaladsl.unmarshalling.Unmarshaller.$anonfun$transform$3(Unmarshaller.scala:23)
	at akka.http.scaladsl.unmarshalling.Unmarshaller$$anon$1.apply(Unmarshaller.scala:58)
	at akka.http.scaladsl.unmarshalling.LowerPriorityGenericUnmarshallers.$anonfun$messageUnmarshallerFromEntityUnmarshaller$3(GenericUnmarshallers.scala:25)
	at akka.http.scaladsl.unmarshalling.Unmarshaller$$anon$1.apply(Unmarshaller.scala:58)
	at akka.http.scaladsl.unmarshalling.Unmarshal.to(Unmarshal.scala:25)
	at skuber.api.client.impl.KubernetesClientImpl.checkResponseStatus(KubernetesClientImpl.scala:669)
	at skuber.api.client.impl.KubernetesClientImpl.toKubernetesResponse(KubernetesClientImpl.scala:645)
	at skuber.api.client.impl.KubernetesClientImpl.$anonfun$makeRequestReturningObjectResource$1(KubernetesClientImpl.scala:188)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
	at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:92)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:92)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:49)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: java.lang.ClassNotFoundException: com.fasterxml.jackson.databind.Module
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 30 more
```

**Comment by skonto** [comment]
Checking a bit deeper the deps we bring in for the current working image vs the one we build from master are as follows:
default:
```
ch.qos.logback.logback-classic-1.2.3.jar
ch.qos.logback.logback-core-1.2.3.jar
com.fasterxml.jackson.core.jackson-annotations-2.9.8.jar
com.fasterxml.jackson.core.jackson-core-2.9.9.jar
com.fasterxml.jackson.core.jackson-databind-2.9.9.3.jar
com.fasterxml.jackson.datatype.jackson-datatype-jdk8-2.9.8.jar
com.fasterxml.jackson.datatype.jackson-datatype-jsr310-2.9.8.jar
com.iheart.ficus_2.12-1.4.7.jar
com.lightbend.akka.akka-stream-alpakka-elasticsearch_2.12-1.1.1.jar
com.lightbend.cloudflow.cloudflow-blueprint-1.3.0-SNAPSHOT.jar
com.lightbend.cloudflow.cloudflow-operator-33-7703ce5.jar
commons-codec.commons-codec-1.11.jar
commons-collections.commons-collections-3.2.1.jar
commons-io.commons-io-2.6.jar
commons-lang.commons-lang-2.6.jar
commons-logging.commons-logging-1.1.3.jar
com.softwaremill.odelay.odelay-core_2.12-0.3.1.jar
com.softwaremill.retry.retry_2.12-0.3.3.jar
com.thoughtworks.paranamer.paranamer-2.7.jar
com.typesafe.akka.akka-actor_2.12-2.5.24.jar
com.typesafe.akka.akka-http_2.12-10.1.3.jar
com.typesafe.akka.akka-http-core_2.12-10.1.3.jar
com.typesafe.akka.akka-parsing_2.12-10.1.3.jar
com.typesafe.akka.akka-protobuf_2.12-2.5.24.jar
com.typesafe.akka.akka-slf4j_2.12-2.5.24.jar
com.typesafe.akka.akka-stream_2.12-2.5.24.jar
com.typesafe.config-1.3.4.jar
com.typesafe.play.play-functional_2.12-2.7.2.jar
com.typesafe.play.play-json_2.12-2.7.2.jar
com.typesafe.ssl-config-core_2.12-0.3.8.jar
io.skuber.skuber_2.12-2.3.0.jar
io.spray.spray-json_2.12-1.3.5.jar
joda-time.joda-time-2.10.1.jar
org.apache.avro.avro-1.8.2.jar
org.apache.avro.avro-compiler-1.8.2.jar
org.apache.commons.commons-compress-1.8.1.jar
org.apache.httpcomponents.httpasyncclient-4.1.2.jar
org.apache.httpcomponents.httpclient-4.5.2.jar
org.apache.httpcomponents.httpcore-4.4.5.jar
org.apache.httpcomponents.httpcore-nio-4.4.5.jar
org.apache.velocity.velocity-1.7.jar
org.bouncycastle.bcpkix-jdk15on-1.60.jar
org.bouncycastle.bcprov-jdk15on-1.60.jar
org.codehaus.jackson.jackson-core-asl-1.9.13.jar
org.codehaus.jackson.jackson-mapper-asl-1.9.13.jar
org.elasticsearch.client.elasticsearch-rest-client-6.3.1.jar
org.reactivestreams.reactive-streams-1.0.2.jar
org.scala-lang.modules.scala-java8-compat_2.12-0.8.0.jar
org.scala-lang.modules.scala-parser-combinators_2.12-1.1.2.jar
org.scala-lang.scala-library-2.12.9.jar
org.scala-lang.scala-reflect-2.12.9.jar
org.slf4j.slf4j-api-1.7.25.jar
org.tukaani.xz-1.5.jar
org.typelevel.macro-compat_2.12-1.1.1.jar
org.xerial.snappy.snappy-java-1.1.1.3.jar
org.yaml.snakeyaml-1.21.jar
```

master:

```
ch.qos.logback.logback-classic-1.2.3.jar
ch.qos.logback.logback-core-1.2.3.jar
com.fasterxml.jackson.core.jackson-annotations-2.9.8.jar
com.fasterxml.jackson.core.jackson-core-2.9.8.jar
com.fasterxml.jackson.datatype.jackson-datatype-jdk8-2.9.8.jar
com.fasterxml.jackson.datatype.jackson-datatype-jsr310-2.9.8.jar
com.iheart.ficus_2.12-1.4.7.jar
com.lightbend.cloudflow.cloudflow-blueprint-1.3.0-M4.jar
com.lightbend.cloudflow.cloudflow-operator-23-efceda0-dirty.jar
commons-codec.commons-codec-1.11.jar
commons-collections.commons-collections-3.2.1.jar
commons-io.commons-io-2.6.jar
commons-lang.commons-lang-2.6.jar
com.thoughtworks.paranamer.paranamer-2.7.jar
com.typesafe.akka.akka-actor_2.12-2.5.24.jar
com.typesafe.akka.akka-http_2.12-10.1.3.jar
com.typesafe.akka.akka-http-core_2.12-10.1.3.jar
com.typesafe.akka.akka-parsing_2.12-10.1.3.jar
com.typesafe.akka.akka-protobuf_2.12-2.5.24.jar
com.typesafe.akka.akka-slf4j_2.12-2.5.24.jar
com.typesafe.akka.akka-stream_2.12-2.5.24.jar
com.typesafe.config-1.3.4.jar
com.typesafe.play.play-functional_2.12-2.7.2.jar
com.typesafe.play.play-json_2.12-2.7.2.jar
com.typesafe.ssl-config-core_2.12-0.3.8.jar
io.skuber.skuber_2.12-2.3.0.jar
io.spray.spray-json_2.12-1.3.5.jar
joda-time.joda-time-2.10.1.jar
org.apache.avro.avro-1.8.2.jar
org.apache.avro.avro-compiler-1.8.2.jar
org.apache.commons.commons-compress-1.8.1.jar
org.apache.velocity.velocity-1.7.jar
org.bouncycastle.bcpkix-jdk15on-1.60.jar
org.bouncycastle.bcprov-jdk15on-1.60.jar
org.codehaus.jackson.jackson-core-asl-1.9.13.jar
org.codehaus.jackson.jackson-mapper-asl-1.9.13.jar
org.reactivestreams.reactive-streams-1.0.2.jar
org.scala-lang.modules.scala-java8-compat_2.12-0.8.0.jar
org.scala-lang.modules.scala-parser-combinators_2.12-1.1.2.jar
org.scala-lang.scala-library-2.12.9.jar
org.scala-lang.scala-reflect-2.12.9.jar
org.slf4j.slf4j-api-1.7.25.jar
org.tukaani.xz-1.5.jar
org.typelevel.macro-compat_2.12-1.1.1.jar
org.xerial.snappy.snappy-java-1.1.1.3.jar
org.yaml.snakeyaml-1.21.jar

```
There are several inconsistencies.

**Comment by skonto** [comment]
Adding the right deps fixes that issue: 
Dependencies:
```val JacksonCore =            "com.fasterxml.jackson.core" % "jackson-core" %"2.9.9"
  val JacksonDataBind           = "com.fasterxml.jackson.core" % "jackson-databind" % "2.9.9.3"
```
Then in build.sbt:
```
lazy val operator =
  cloudflowModule("cloudflow-operator")
    .enablePlugins(
      sbtdocker.DockerPlugin,
      JavaAppPackaging,
      BuildNumberPlugin,
      BuildInfoPlugin
    )
    .dependsOn(blueprint % "compile->compile;test->test")
    .settings(
      libraryDependencies ++= Vector(
        AkkaSlf4j,
        AkkaStream,
        Ficus,
        Logback,
        Skuber,
        AkkaStreamTestkit,
        JacksonDataBind,
        JacksonCore,
        ScalaTest,
        ScalaCheck              % "test",
        Avro4sJson              % "test",
      )
    )```

Will test and create a PR for this.

**Comment by agemooij** [comment]
Fixed by #44
